platforms/software/dependency-management/src/integTest/groovy/org/gradle/plugin/devel/variants/TargetJVMVersionOnPluginTooNewFailureDescriberIntegrationTest.groovy

```diff
@@ -0,0 +1,197 @@
+    Integer currentJava = Integer.valueOf(JavaVersion.current().majorVersion)
+    Integer tooHighJava = currentJava + 1
+
+    def 'JVM version too low uses custom error message for plugin'() {
+        given:
+        def producer = file('producer')
+        def consumer = file('consumer')
+        def pluginModule = mavenRepo.module('com.example', 'producer', '1.0')
+        def pluginMarker = mavenRepo.module('com.example.greeting', 'com.example.greeting.gradle.plugin', '1.0')
+
+        producer.file('settings.gradle').createFile()
+        producer.file('build.gradle') << """
+            plugins {
+                id('java-gradle-plugin')
+                id('maven-publish')
+            }
+
+            group = "com.example"
+            version = "1.0"
+
+            configurations.configureEach {
+                if (canBeConsumed)  {
+                    attributes {
+                        attribute(TargetJvmVersion.TARGET_JVM_VERSION_ATTRIBUTE, $tooHighJava)
+                    }
+                }
+            }
+
+            gradlePlugin {
+                plugins.create('greeting') {
+                    id = 'com.example.greeting'
+                    implementationClass = 'example.plugin.GreetingPlugin'
+                }
+            }
+            publishing {
+                repositories {
+                    maven { url = '${mavenRepo.uri}' }
+                }
+            }
+        """
+        producer.file('src/main/java/example/plugin/GreetingPlugin.java') << pluginImplementation()
+
+        consumer.file('settings.gradle') << """
+            pluginManagement {
+                repositories {
+                    maven { url = '${mavenRepo.uri}' }
+                }
+            }
+        """
+        consumer.file('build.gradle') << """
+            plugins {
+                id('com.example.greeting') version '1.0'
+            }
+        """
+
+        when:
+        projectDir(producer)
+        succeeds 'publish'
+
+        then:
+        pluginModule.assertPublished()
+        pluginMarker.assertPublished()
+        pluginModule.artifact([:]).assertPublished()
+
+        when:
+        projectDir(consumer)
+        fails 'greet', "--stacktrace"
+
+        then:
+        failure.assertHasErrorOutput("""> Could not resolve all artifacts for configuration ':classpath'.
+   > Could not resolve com.example:producer:1.0.
+     Required by:
+         project : > com.example.greeting:com.example.greeting.gradle.plugin:1.0
+      > Plugin com.example:producer:1.0 requires at least a Java $tooHighJava JVM. This build uses a Java $currentJava JVM.""")
+        failure.assertHasErrorOutput("Caused by: " + VariantSelectionException.class.getName())
+        failure.assertHasResolution("Run this build using a Java $tooHighJava JVM (or newer).")
+    }
+
+    @Requires(UnitTestPreconditions.Jdk11OrEarlier)
+    def 'spring boot 3 plugin usage with jvm < 17 uses custom error message'() {
```

Comment:
🤔 Can we keep this in the smoke tests? I know it's a little more specific about our behavior, but it seems right that we should keep this closer to where we test against spring boot elsewhere.

Alternatively, I think it's fine to rely on the other test we have here and not test against a real plugin.